src/services/navigationBar.ts

```diff
@@ -2,856 +2,551 @@
-                const childItems = getItemsWorker(sortNodes(removeDynamicallyNamedProperties(node)), createChildItem);
-                return getNavigationBarItem(
-                    node.name.text,
-                    ts.ScriptElementKind.interfaceElement,
-                    getNodeModifiers(node),
-                    [getNodeSpan(node)],
-                    childItems,
-                    getIndent(node));
-            }
+            target.children.push(...source.children);
+            mergeChildren(target.children);
+            sortChildren(target.children);
         }
+    }
 
-        function getModuleName(moduleDeclaration: ModuleDeclaration): string {
-            // We want to maintain quotation marks.
-            if (isAmbientModule(moduleDeclaration)) {
-                return getTextOfNode(moduleDeclaration.name);
+    /** Recursively ensure that each NavNode's children are in sorted order. */
+    function sortChildren(children: NavNode[]): void {
+        children.sort((child1, child2) => {
+            const name1 = tryGetName(child1.node), name2 = tryGetName(child2.node);
+            if (name1 && name2) {
+                const cmp = localeCompareFix(name1, name2);
+                return cmp !== 0 ? cmp : navKind(child1) - navKind(child2);
             }
-
-            // Otherwise, we need to aggregate each identifier to build up the qualified name.
-            const result: string[] = [];
-
-            result.push(moduleDeclaration.name.text);
-
-            while (moduleDeclaration.body && moduleDeclaration.body.kind === SyntaxKind.ModuleDeclaration) {
-                moduleDeclaration = <ModuleDeclaration>moduleDeclaration.body;
-
-                result.push(moduleDeclaration.name.text);
+            else {
+                return name1 ? 1 : name2 ? -1 : navKind(child1) - navKind(child2);
             }
+        });
+    }
 
-            return result.join(".");
+    // node 0.10 treats "a" as greater than "B".
+    const localeCompareIsCorrect = "a".localeCompare("B") < 0;
+    function localeCompareFix(a: string, b: string): number {
+        if (localeCompareIsCorrect) {
+            return a.localeCompare(b);
         }
-
-        function removeComputedProperties(node: EnumDeclaration): Declaration[] {
-            return filter<Declaration>(node.members, member => member.name === undefined || member.name.kind !== SyntaxKind.ComputedPropertyName);
+        else {
+            // This isn't perfect, but it passes all of our tests.
+            for (let i = 0; i < Math.min(a.length, b.length); i++) {
+                const chA = a.charAt(i), chB = b.charAt(i);
+                if (chA === "\"" && chB === "'") {
+                    return 1;
+                }
+                if (chA === "'" && chB === "\"") {
+                    return -1;
+                }
+                const cmp = chA.toLowerCase().localeCompare(chB.toLowerCase());

```

Comment:
If we're using `localeCompare`, we should probably use `toLocaleLowerCase` for lower-casing things in a locale-sensitive way.